File: Forge/src/main/java/siongsng/rpmtwupdatemod/gui/widget/TranslucentButton.java
package siongsng.rpmtwupdatemod.gui.widget;


public class TranslucentButton extends RPMButton {

    public TranslucentButton(int x, int y, int width, int height, String message, RPMButton.IPressable onPress, String tooltip) {
        super(x, y, width, height, message, onPress, tooltip);
    }

    @Override
    public void drawTexturedModalRect(int x, int y, int textureX, int textureY, int width, int height) {
        drawRect(x, y, x + width, y + height, Integer.MIN_VALUE);
    }

}

File: common/src/main/kotlin/com/rpmtw/rpmtw_platform_mod/gui/widgets/TranslucentButton.kt
package com.rpmtw.rpmtw_platform_mod.gui.widgets

import com.mojang.blaze3d.vertex.PoseStack
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.components.Button
import net.minecraft.network.chat.Component

class TranslucentButton(
    x: Int,
    y: Int,
    width: Int,
    height: Int,
    message: Component,
    onPress: OnPress,
    val tooltip: Component?,
) : Button(x, y, width, height, message, onPress) {
    override fun renderButton(poseStack: PoseStack, i: Int, j: Int, f: Float) {
        super.renderButton(poseStack, i, j, f)
        if (this.isHoveredOrFocused && tooltip != null) {
            Minecraft.getInstance().screen?.renderTooltip(
                poseStack,
                tooltip,
                i,
                j
            )
        }
    }
}